plugin/trino-hive/src/main/java/io/trino/plugin/hive/PartitionUpdate.java

```diff
@@ -163,7 +164,7 @@ public HiveBasicStatistics getStatistics()
         return new HiveBasicStatistics(fileNames.size(), rowCount, inMemoryDataSizeInBytes, onDiskDataSizeInBytes);
     }
 
-    public static List<PartitionUpdate> mergePartitionUpdates(Iterable<PartitionUpdate> unMergedUpdates)
+    public static List<PartitionUpdate> mergePartitionUpdates(Iterator<PartitionUpdate> unMergedUpdates)
```

Comment:
again: refactorint  for this to consume iterator should be a preparatory commit.

Reply:
done